library/src/main/java/com/feedhenry/sdk/Sync.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright Red Hat, Inc, and individual contributors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.feedhenry.sdk;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import com.feedhenry.sdk.android.ClientIdGeneraorImpl;
+import com.feedhenry.sdk.android.FileStorageImpl;
+import com.feedhenry.sdk.android.NetworkClientImpl;
+import com.feedhenry.sdk.android.SyncableActivity;
+import com.feedhenry.sdk.network.NetworkClient;
+import com.feedhenry.sdk.sync.FHSyncClient;
+import com.feedhenry.sdk.sync.FHSyncConfig;
+import okhttp3.MediaType;
+
+/**
+ * The Sync class provides static methods to initialize the library, create new instances of all the
+ * API request objects, and configure global settings.
+ */
+public class Sync {
+
+    private static final String TAG = "FHSync";
+
+    public static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+
+    public static final int LOG_LEVEL_VERBOSE = 1;
+    public static final int LOG_LEVEL_DEBUG = 2;
+    public static final int LOG_LEVEL_INFO = 3;
+    public static final int LOG_LEVEL_WARNING = 4;
+    public static final int LOG_LEVEL_ERROR = 5;
+    public static final int LOG_LEVEL_NONE = Integer.MAX_VALUE;
+
+    private static int mLogLevel = LOG_LEVEL_ERROR;
+
+    private static NetworkClient networkClient;
+
+    private Sync() {
+    }
+
+    /**
+     * Sync initialization.
+     * @param application application object
+     * @param config sync config
+     * @param cloudUrl url of the cloud app endpoint
+     */
+    public static void init(final Application application, final FHSyncConfig config, final String cloudUrl) {
+        final Context appContext = application.getApplicationContext();
+        networkClient = new NetworkClientImpl(appContext,cloudUrl,new ClientIdGeneraorImpl(appContext));
+        networkClient.registerNetworkListener();
+        application.registerActivityLifecycleCallbacks(new Application.ActivityLifecycleCallbacks() {
+            @Override
+            public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+                if (activity instanceof SyncableActivity) {
```

Comment:
does this mean `SyncableActivity` only works with `Activities`? What if people want to use it inside `Fragments`?